docs/tutorials/leaderboards.md

```diff
@@ -177,6 +177,90 @@ The message is just a basic message to inform you to the status the download; wh
 - **ugc_handle:** handle for any UGC that is attached to this entry
 - **details:** any details you stored with this entry for later use
 
+{==
+## Uploading UGC to Leaderboard
+==}
+
+You can use this method if you want to attach replays or any other data to leadearboard entry.
+So first you will need to setup signals
+
+=== "Godot 2.x, 3.x"
+
+	```gdscript
+	Steam.connect("file_share_result", self, "_on_file_share_result")
+	Steam.connect("file_share_result", self, "on_file_write_async_complete")
+	Steam.connect("leaderboard_ugc_set", self, "on_leaderboard_ugc_set")
+	```
+
+=== "Godot 4.x"
+
+	```
+	Steam.connect("file_share_result", Callable(self, "on_file_share_result"))
+	Steam.connect("file_write_async_complete", Callable(self, "on_file_write_async_complete"))
+	Steam.connect("leaderboard_ugc_set", Callable(self, "on_leaderboard_ugc_set"))
+	```
+	
+### Creating a file we want to share
+
+Created file is stored in PackedByteArray format.
+Frist we need to create a file with content that we want to share.
+
+```gdscript
+var data:PackedByteArray
```

Comment:
For readability, spaces should be after `data:` and all arguments in the `fileWriteAsync` call.

Reply:
done